### PR TITLE
Bump TFX to 0.21.1

### PR DESCRIPTION
**Pull request by numerology**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3220)

### CODE REVIEW

**Comment by numerology** [comment]
/retest

One more try then I'm going to take a look at GKE setup

**Comment by rmgogogo** [comment]
/cc Bobgy as you changed TFMA

Is it tested locally? standalone CLI can help a quick test

**Comment by numerology** [comment]
> /cc Bobgy as you changed TFMA
> 
> Is it tested locally? standalone CLI can help a quick test

It looks okay at https://465b70d4a3050db6-dot-us-central2.pipelines.googleusercontent.com/#/runs/details/4cc460ce-1e58-4c77-99fb-c9ee5110a2b1

Per offline discussion there'll be some further change:

-[ ] Deprecate ModelValidator
-[ ] Add a keras native sample.

File: test/deploy-cluster.sh [hunk 1]
```diff
@@ -69,7 +69,7 @@ else
   NODE_POOL_CONFIG_ARG="--num-nodes=2 --machine-type=n1-standard-8 \
     --enable-autoscaling --max-nodes=8 --min-nodes=2"
   # Use new kubernetes master to improve workload identity stability.
-  KUBERNETES_VERSION_ARG="--cluster-version=1.14.8-gke.33"
+  KUBERNETES_VERSION_ARG="--cluster-version=1.14.10-gke.17"
```
Comment (Bobgy) [inline test/deploy-cluster.sh]: Maybe we can just specify 1.14?

Comment (numerology) [reply test/deploy-cluster.sh]: Good idea

**Comment by numerology** [comment]
> /lgtm
> Are these samples covered in presubmit tests?

Yes they are.

**Comment by Bobgy** [comment]
Right, @numerology Can you verify if TFMA/TFDV visualizations are still working?
Unfortunately, we don't have e2e test coverage yet

**Comment by Bobgy** [comment]
UPDATE: discussed offline, verified they still work

**Comment by numerology** [comment]
UPDATE: will update the GKE master version in the following PRs

**Pull request merged**